### PR TITLE
Add documentation for AsEventListener attribute

### PR DESCRIPTION
**Pull request by jderusse**
This PR adds documentation for `AsEventListener` attribute introduced in #1561

### CODE REVIEW

**Comment by ostrolucky** [comment]
can you update doc/ too?

**Comment by jderusse** [comment]
Documentation added here (DoctrineBundle) and also in symfony/symfony-docs https://github.com/symfony/symfony-docs/pull/17472

**Comment by nicolas-grekas** [comment]
What about naming this one `AsOrmListener`? Symfony already has an `AsEvenListener` attribute and that could be confusing.

**Review by ostrolucky — APPROVED** [review]

**Comment by ostrolucky** [comment]
This confusion just reflects how it is in DI tags, so it's fine I think. If we name this AsOrmListener, people would then wonder why tag isn't doctrine.orm.event_listener

**Comment by dmaicher** [comment]
And actually `doctrine.event_listener` is not purely an ORM concept, or? I think this will also work for DBAL events.

File: Attribute/AsEventListener.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\Attribute;
+
+use Attribute;
+
+/**
+ * Service tag to autoconfigure entity listeners.
```
Comment (dmaicher) [inline Attribute/AsEventListener.php]:
```suggestion
 * Service tag to autoconfigure event listeners.
```

File: Resources/doc/event-listeners.rst [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+In opposite to :doc:`Entity Listeners </entity-listeners>`, Events listeners
+are services that listen for all entities in your application.
+
+See https://www.doctrine-project.org/projects/doctrine-orm/en/latest/reference/events.html#implementing-event-listeners
+for more info on events listeners.
+
+To register a service to act as an event listener you have to tag it with the
+``doctrine.event_listener`` tag:
+
+Starting with Doctrine bundle 2.7, you can use the ``AsEventListener``
```
Comment (dmaicher) [inline Resources/doc/event-listeners.rst]:
```suggestion
Starting with Doctrine bundle 2.8, you can use the ``AsEventListener``
```

**Review by dmaicher — COMMENTED** [review]
:+1: 

Regarding the name: I agree its slightly confusing with Symfony's `AsEventListener`. Maybe `AsDoctrineEventListener` could be an alternative? :man_shrugging:

**Comment by dmaicher** [comment]
@jderusse can you change the base to 2.8.x? Since this is a new feature we should target 2.8

**Comment by jderusse** [comment]
ok... this is embarrassing now... While rebasing on `2.8.x` I just realized this is already implemented in #1561....

I keep this PR open for documentation purposes

File: Resources/doc/event-listeners.rst [hunk 3]
```diff
@@ -0,0 +1,103 @@
+Event Listeners
+===============
+
+In opposite to :doc:`Entity Listeners </entity-listeners>`, Events listeners
```
Comment (dmaicher) [inline Resources/doc/event-listeners.rst]:
```suggestion
In opposite to :doc:`Entity Listeners </entity-listeners>`, Event listeners
```

File: Resources/doc/event-listeners.rst [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+are services that listen for all entities in your application.
+
+See https://www.doctrine-project.org/projects/doctrine-orm/en/latest/reference/events.html#implementing-event-listeners
+for more info on events listeners.
```
Comment (dmaicher) [inline Resources/doc/event-listeners.rst]:
```suggestion
for more info on event listeners.
```

File: Resources/doc/event-listeners.rst [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            $services->set(SearchIndexer::class)
+                ->tag('doctrine.event_listener', [
+                    // this is the only required option for the lifecycle listener tag
+                    'event' => 'postPersist',
+
+                    // listeners can define their priority in case multiple subscribers or listeners are associated
+                    // to the same event (default priority = 0; higher numbers = listener is run earlier)
+                    'priority' => 500,
+
+                    # you can also restrict listeners to a specific Doctrine connection
```
Comment (dmaicher) [inline Resources/doc/event-listeners.rst]:
```suggestion
                    // you can also restrict listeners to a specific Doctrine connection
```

**Review by dmaicher — COMMENTED** [review]
Oh wow I also totally missed that this was already done on 2.8 :see_no_evil: 

Some tiny remarks left from my side :+1:

**Comment by nicolas-grekas** [comment]
I'd like we rename the attribute, here is my proposal: https://github.com/doctrine/DoctrineBundle/pull/1592

**Comment by dmaicher** [comment]
@jderusse the attribute was renamed to `AsDoctrineListener` now. Are you up for adjusting the documentation?

**Comment by ostrolucky** [comment]
I've updated it. Any other feedback?

**Review by dmaicher — APPROVED** [review]

**Pull request merged**